migrations/2023-12-18-114245_fastn_auth_tables/down.sql

```diff
@@ -0,0 +1,11 @@
+-- This file should undo anything in `up.sql`
+
+DROP TABLE IF EXISTS fastn_email_confirmation;
+DROP TABLE IF EXISTS fastn_user_email;
+DROP TABLE IF EXISTS fastn_oauthtoken;
+DROP TABLE IF EXISTS fastn_session;
+DROP TABLE IF EXISTS fastn_user;
+
+DROP DOMAIN IF EXISTS email;
+DROP EXTENSION IF EXISTS citext;
```

Comment:
Checkout the TIP in https://www.postgresql.org/docs/current/citext.html#id-1.11.7.19.9:

> Consider using nondeterministic collations (see [Section 24.2.2.4](https://www.postgresql.org/docs/current/collation.html#COLLATION-NONDETERMINISTIC)) instead of this module.

They are discouraging using of `citext` now, but I am not sure what exactly they are recommending, like how to apply their tip in concrete manner. Do read it up and let me know if it makes sense to you.

Reply:
The `COLLATE` features seems nice and they've provided some examples too: (including case-insensitive comparisons) [https://www.postgresql.org/docs/current/collation.html#COLLATION-NONDETERMINISTIC](postgresql.org/docs/current/collation.html#COLLATION-NONDETERMINISTIC)

It seems complex to implement than the current `citext` we already have. I am going to ignore this for now as I don't fully understand the consequences of locale handling (I doubt there are any in this case)